### DISCUSSION TITLE
OnTap is triggered automatically

### DISCUSSION THREAD

**Discussion by MALLORY8K** [darioielardi/flutter_speed_dial#115] [Q&A] [answered]
when I go to my site OnTap is triggered automatically! how i can fix this?

**Comment by prateekmedia** [comment]
> when I go to my site OnTap is triggered automatically! how i can fix this?

Can you provide more info please, I dont understand your problem.

Maybe link some code that you used for speed dial and also the version of flutter_speed_dial you are using from the pubspec.yaml

**Comment by MALLORY8K** [comment]
i Have a function and call this onTap but, if i start the app, the function is call  automatically.

```dart
myFunction(){
   print('Hello');
}

SpeedDialChild(
   child: Icon(Icons.brush),
   backgroundColor: Colors.blue,
   label: 'Second',
   labelStyle: TextStyle(fontSize: 18.0),
   onTap: myFunction(),
),
```

**Comment by prateekmedia** [comment, marked as answer]
@MarcelHH 
Firstly, convert 
> onTap: myFunction(),

to

> onTap: myFunction,

as it only accepts VoidCallbacks and not a function as a whole.

Secondly it doesn't seems related to this library rather it seems you are calling this function somewhere else in your code, you can easily find it by searching for

> myFunction()

If you find the error then you can close this issue, else you can write what you did and what didn't work.

**Comment by MALLORY8K** [comment]
onTap: myFunction,

that solved my problem thank you
